src/components/PostsList/PostsList.jsx

```diff
@@ -1,37 +1,87 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import './PostsList.scss';
 
-export const PostsList = () => (
+export const PostsList = ({
+  posts,
+  selectedUser,
+  selectedPost,
+  setSelectedPost,
+  isOpen,
+  setIsOpen,
+  setAreCommentsHidden,
+}) => (
   <div className="PostsList">
     <h2>Posts:</h2>
 
     <ul className="PostsList__list">
-      <li className="PostsList__item">
-        <div>
-          <b>[User #1]: </b>
-          sunt aut facere repellat provident occaecati excepturi optio
-        </div>
-        <button
-          type="button"
-          className="PostsList__button button"
-        >
-          Close
-        </button>
-      </li>
+      {posts.map((post) => {
+        if (
+          selectedUser === '0'
+            || +selectedUser === post.userId
+        ) {
```

Comment:
better to prepare your array with posts somewhere else and mapping here already prepared posts